src/ConfigPostProcessor.php

```diff
@@ -265,19 +260,82 @@ private function replaceDependencyConfiguration(array $config)
      *
      * In this case, we want to keep the alias as-is, but rewrite the target.
      *
-     * This same logic can be used for invokables, which are essentially just
-     * an alias map.
+     * We need also provide an additional alias if the alias key is a legacy class.
      *
      * @return array
      */
     private function replaceDependencyAliases(array $aliases)
     {
         foreach ($aliases as $alias => $target) {
-            $aliases[$alias] = $this->replacements->replace($target);
+            $newTarget = $this->replacements->replace($target);
+            $newAlias  = $this->replacements->replace($alias);
+
+            if ($newAlias === $alias) {
+                $aliases[$alias] = $newTarget;
+                continue;
+            }
+
+            if (isset($aliases[$newAlias])) {
+                continue;
+            }
+
+            $notIn = [$newTarget];
+            $name  = $newTarget;
+            while (isset($aliases[$name])) {
+                $notIn[] = $aliases[$name];
+                $name    = $aliases[$name];
+            }
+
+            if (! in_array($newAlias, $notIn, true)) {
+                $aliases[$alias]    = $newAlias;
+                $aliases[$newAlias] = $newTarget;
+            }
         }
+
         return $aliases;
     }
 
+    /**
+     * Rewrite dependency invokables array
+     *
+     * In this case, we want to keep the alias as-is, but rewrite the target.
+     *
+     * We need also provide an additional alias if invokable is defined with
+     * an alias which is a legacy class.
+     *
+     * @return array
+     */
+    private function replaceDependencyInvokables(array $config)
+    {
+        if (empty($config['invokables'])) {
+            return $config;
+        }
+
+        foreach ($config['invokables'] as $alias => $target) {
+            $newTarget = $this->replacements->replace($target);
+
+            if ($alias === $target) {
+                $config['invokables'][$alias] = $newTarget;
+                continue;
+            }
+
+            $newAlias = $this->replacements->replace($alias);
+            $config['invokables'][$newAlias] = $newTarget;
+
+            if ($newAlias === $alias) {
+                continue;
+            }
+
+            if (! isset($config['aliases'][$newAlias])) {
```

Comment:
@weierophinney I think something might be wrong here - just checked and if I change this line to:
```php
if (! isset($config['aliases'][$alias])) {
```
all tests are still green. I think:
- we are missing some test case
- the above condition is invalid and should be as provided above in that comment.

Reply:
So, we need test cases! Figure out what's missing from the tests, see if they continue to pass with the code as-is, and then changing that condition.